src/main.rs

```diff
@@ -1,52 +1,59 @@
+use jinko;
 use yew::prelude::*;
 
 enum Msg {
-    Update(String),
+    Update(Option<String>),
     Eval,
 }
 
 struct Model {
     // `ComponentLink` is like a reference to a component.
     // It can be used to send messages to the component
-    link: ComponentLink<Self>,
+    // link: ComponentLink<Self>,
     input: String,
     output: Vec<String>,
+    jk_context: jinko::Context,
+}
+
+#[derive(Clone, Copy)]
+struct Reader;
+impl jinko::JkReader for Reader {
+    fn read_to_string(&self, path: &str) -> Result<String, jinko::Error> {
+        panic!("Read is not implemented, tried to be called on {}", path);
+    }
 }
 
 impl Component for Model {
     type Message = Msg;
     type Properties = ();
 
-    fn create(_props: Self::Properties, link: ComponentLink<Self>) -> Self {
+    fn create(_ctx: &Context<Self>) -> Self {
         Self {
-            link,
             input: String::new(),
             output: Vec::new(),
+            jk_context: jinko::Context::new(Box::new(Reader {})),
```

Comment:
```suggestion
            jk_context: jinko::Context::new(Box::new(Reader)),
```

Reply:
You should be able to use value structs like so

Reply:
Fixed